src/components/LogRow/BaseRow/SharingMenu/index.tsx

```diff
@@ -0,0 +1,154 @@
+    }
+    setOpen(!open);
+  };
+
+  const handleCopySelectedLines = async () => {
+    const { endingLine, startingLine } = selectedLines;
+    if (startingLine === undefined || endingLine === undefined) return;
+    // Create an array of line numbers that represent the range in selectedLines
+    const lineNumbers = getLinesInProcessedLogLinesFromSelectedLines(
+      processedLogLines,
+      selectedLines
+    );
+
+    await copyToClipboard(getJiraFormat(lineNumbers, getLine));
+    setOpen(false);
+    sendEvent({
+      name: "Copied Share Lines To Clipboard",
+    });
+    dispatchToast.success(`Copied ${lineNumbers.length} lines to clipboard`);
+  };
+
+  const handleOnlySearchOnRange = () => {
+    const { endingLine, startingLine } = selectedLines;
+    if (startingLine === undefined || endingLine === undefined) return;
+    setParams({
+      ...params,
+      [QueryParams.LowerRange]: startingLine,
+      [QueryParams.UpperRange]: endingLine,
+    });
+    setOpen(false);
+    sendEvent({
+      name: "Applied Range Limit",
+    });
+  };
+
+  const handleShareLinkToSelectedLines = async () => {
+    const { startingLine } = selectedLines;
+    if (startingLine === undefined) return;
+    // Take the current URL and add the shareLine query param
+    const url = new URL(window.location.href);
+    url.searchParams.set(QueryParams.ShareLine, startingLine.toString());
+
+    await copyToClipboard(url.toString());
+    setOpen(false);
+    sendEvent({ name: "Copied Share Link" });
+    dispatchToast.success("Copied link to clipboard");
+  };
+
+  const lineCount =
+    selectedLines.endingLine === undefined ||
+    selectedLines.startingLine === undefined
+      ? 1
+      : selectedLines.endingLine - selectedLines.startingLine + 1;
+
+  return (
+    <StyledMenu
+      align="right"
+      data-cy="sharing-menu"
+      open={open}
+      setOpen={setMenuOpen}
+      trigger={
+        <MenuIcon
+          aria-label="Expand share menu"
+          data-cy="sharing-menu-button"
+          onClick={setMenuOpen}
+        >
+          <Icon glyph="Ellipsis" />
+        </MenuIcon>
+      }
+    >
+      <MenuItem
+        glyph={<Icon glyph="Copy" />}
+        onClick={handleCopySelectedLines}
+        title={`Copy the selected ${pluralize(
+          "line",
+          lineCount
+        )} to your clipboard with JIRA formatting.`}
+      >
+        Copy selected contents
+      </MenuItem>
+      {!isUploadedLog && (
+        <MenuItem
+          glyph={<Icon glyph="Export" />}
+          onClick={handleShareLinkToSelectedLines}
+          title={`Copy a link to ${pluralize("this", lineCount)} ${pluralize(
+            "line",
+            lineCount
+          )}.`}
+        >
+          Copy share link to selected {pluralize("line", lineCount)}
+        </MenuItem>
+      )}
+      <MenuItem
+        disabled={lineCount === 1}
+        glyph={<Icon glyph="MagnifyingGlass" />}
+        onClick={handleOnlySearchOnRange}
```

Comment:
Sophie also suggested the same thing above. I have a follow-up ticket to keep this pr smaller. https://jira.mongodb.org/browse/DEVPROD-1647 I will work on it immediately after merging this since this PR is already pretty complex.

Reply:
I don't consider "search on range" to be a filter instead it's more so the logic associated with how we perform searches in the app. We don't apply that logic to any of the other functionality located in the details menu so I don't necessarily thing search on range should be an exception. 
<img width="741" alt="image" src="https://github.com/evergreen-ci/parsley/assets/4605522/9ffe161c-3e7c-4b27-ba44-93d01d288bd9">

Reply:
If we do want to change this behavior we should apply the concept universally and it would probably be out of scope for this ticket since it would also affect how we render the navbar since its already pretty cramped.
<img width="1494" alt="image" src="https://github.com/evergreen-ci/parsley/assets/4605522/0a596f29-d903-4eef-bb55-ec79ff8b73e1">